packages/parchment/lib/src/convert/html.dart

```diff
@@ -205,26 +211,43 @@ class _ParchmentHtmlEncoder extends Converter<Delta, String> {
     }
 
     // Close any remaining blocks
+    _closeOpenBlocks(state);
+
+    // Remove default paragraph block if single line of text
+    String result = state.buffer.toString();
+    if (state.isSingleLine && result.startsWith('<p>')) {
+      result = result.substring('<p>'.length, result.length - '</p>'.length);
+    }
+
+    // Remove the final <br> if there is one.
+    result = result
+        .replaceFirst(_brPrEolRegex, '</p>')
+        .replaceFirst(_brEolRegex, '');
+    return result;
+  }
+
+  /// Closes all open blocks and returns the ending position.
+  int _closeOpenBlocks(_EncoderState state) {
```

Comment:
This only works for checklist followed by paragraph but doesn't work for other blocks followed by paragraph.
I've come up with this
```dart
int _closeOpenBlocks(_EncoderState state,
      {bool beforePlainParagraphHandling = false}) {
    final openBlockTags = state.openBlockTags;
    final buffer = state.buffer;
    final numToClose = openBlockTags.length;
    int position = 0;
    for (var i = 0; i < numToClose; i++) {
      final blockTag = openBlockTags[i];
      if (position > 0) {
        blockTag.closingPosition = position;
      } else {
        position = blockTag.closingPosition;
      }
      if (!isPlain(blockTag.style)) {
        position += blockTag.inducedPadding;
      }
      if (i == numToClose - 1 && !beforePlainParagraphHandling) {
        blockTag.closingPosition = buffer.length;
      }
      _writeBlockTag(buffer, blockTag);
    }

    state.openBlockTags.clear();
    return numToClose == 1 ? position : buffer.length;
  }
```

for the last opened tags, if a paragraph follows, we will be using the blockTag's closing position.

Also there were problems with nested block, the position variable is updated to handle this

Reply:
OK, all the tests were passing with my version. Perhaps you need a test to demonstrate your failing case? Nevertheless, I put your version in and all of the still pass.

Reply:
Yes this failed
```json
{
  "input":[
    {"insert": "some code"},
    {
      "insert": "\n",
      "attributes": {"block": "code"}
    },
    {"insert": "Hello world", "attributes":{"b": true}},
    {"insert": "\n"}
  ]
}
```
```dart
'<pre><code>some code\n</code></pre><p><strong>Hello world</strong></p>'
```

Reply:
Great! I've added a test.